specification/network/resource-manager/Microsoft.Network/stable/2023-04-01/firewallPolicy.json

```diff
@@ -355,6 +355,248 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/firewallPolicies/{firewallPolicyName}/putDraft": {
```

Comment:
>putDraft [](http://example.com/codeflow?start=138&length=8)

[ARM blocking signoff] This is really not recommended. You have implemented the put\deploy\get\delete as post actions for the draft resource. This really seems like it should have instead been modelled as resources that can be PUT\GET\DELETE like normal resource. WHy have you chosen to have these modelled as POST operations instead ?

Reply:
I can get an official statement from our PM / Lead on this, but it was because we didn't want it to be a resource itself. It's not tracked or necessarily impacting. It's meant to be either a "safe" method or a draft.

I will contact you internally on implementation

Reply:
Why was this API chosen? 

Backwards compatibility to current API so we don’t need to create a new version. 
We preferred not creating new ARM resources because that would possibly create RBAC problem. 
Creating new ARM resources would also be problematic during deploy draft because we would need to delete all the relevant draft resources (policy and rule collection groups) in one call.

Reply:
I am not sure I understand. 
- You are literally creating a new api version with new resources so it is recommended to follow the guidelines correctly. POST is not recommended for creation of ARM resources. This would mean that ARM tools \ polciy \ARN , will not work as expected for these resources 
- What is the RBAC problem with creating new resources ? Child resources inherit the RBAC of thier parents if they are proxy, so I am not sure what the problem here is ?
- I dont really follow this argument. You still need to call deleteDraft multiple times right ?

Reply:
Hey @raosuhas,

1. Draft has 1:1 relationship with its policy or RCG. Having it as sub resource (which usually has 1:many relationship with their parent) does not make much sense and can confuse the Cx.

2. By the nature of the deploy draft action (on the policy) we remove the entire set of drafts that is related to the policy (1 draft policy and possibly many draft RCGs). Having all those as resources, will require us to delete all those draft resources upon policy 'deploy' action and this is not a common and suggested practice as far as we understand. 
In addition, if drafts will be resources, their structure will be ../policy/policyDraft and ../policy/rcg/rcgDraft. This structure implies that rcgDraft resource has no relation to policyDraft resource which does align with the fact that an action on the policy draft triggers a deletion of the rcg drafts.

An option we can go with is change the naming of put/get/delete draft to something else (e.g store/read/remove) to differentiate from the common resource actions.

I suggest to have a meeting to discuss about this to find the best option. Please suggest availability and right forum.
Thanks!